documentation/src/main/asciidoc/topics/ref_performance_tuning.adoc

```diff
@@ -79,6 +79,90 @@ The recommended values are:
+|`cached-thread-pool` |string| A thread pool executor that creates new threads on demand. The `name` and `thread-factory` attributes are mandatory.
+
+|`non-blocking-bounded-queue-thread-pool` | string and integer | A thread pool executor that can run non-blocking operations. 
+Executor creates new threads when the number of running threads is less that the core size. 
+Idle threads are placed in a queue.
+
+Includes similar attributes to the `blocking-bounded-queue-thread-pool` component.
+
+|`scheduled-thread-pool`|string|A thread pool executor that creates a single-threaded executor that schedules commands to run at a specific time.
+The `name` and `thread-factory` attributes are mandatory. 
+
+|`thread-factory` |string| Specifies metadata for your threads pools. 
+The `name` attribute and the `thread-name-pattern` attributes are mandatory. 
+Optional attributes include `group-name` and `priority`.
+|===
+
+Your {brandname} configuration contains a cache container security configuration that applies default settings to thread pools.
+You can change these settings to meet your needs.
+
+The following table outlines common thread pool properties and their default thread property settings that exist in the cache container security configuration. 
+Each thread pool property contains a `DEFAULT_THREAD_PRIORITY` thread property that default to `NORM_PRIORITY`.   
+
+[cols="1,2,3"]
+|===
+| Property | `DEFAULT_THREAD_COUNT` / `DEFAULT_QUEUE_SIZE` | Description
+
+|`ASYNC_NOTIFICATION_EXECUTOR` |`1` / `1_000`| Used by asynchronous listeners that your {brandname} server registers in embedded mode. 
+The default setting ensures that Infinispan invokes asynchronous listeners according to importance of their messages. 
+Configuring this property is discouraged.
+Increasing the property's value from `1` might cause your {brandname} server to randomly create mixed types of listener events on different threads.
+
+|`BLOCKING_EXECUTOR` | `150` / `5_000`  |  Used by your {brandname} server to handle blocking operations.
+An example of a blocking operation includes a cache store that calls external sources, transactions, or time-consuming non-blocking entities.
+Thread pool reuses blocking threads based on the value specified for this property.
+For workloads with large amounts of blocking threads, you set a higher value for the `BLOCKING_EXECUTOR` property.
+
+|`EXPIRATION_SCHEDULED_EXECUTOR`|`1` / `0`| Based on the property's value, a reaper schedules checks on expired cache entries that exist in memory.
+If your workload includes a large amount of expired cache entries that span multiple caches, consider setting a higher value for this property. 
+Be mindful that setting a higher value does not increase the rate at which threads process expired cache entries. 
+The default setting of the `EXPIRATION_SCHEDULED_EXECUTOR` property causes one thread to handle all cache entries for a single cache.
```

Comment:
default to 1 thread, but I struggle with "single cache".
One means that there is one thread handle all caches one after the other to remove all expired entries.
More means it can handle X caches in parallel

Reply:
Agreed, this could be a bit more clear.

Reply:
I moved an updated note to the `non-blocking-bounded-queue-thread-pool` table entry. WDYT?

> Note: If you have a large amount of running CPU cores in your environment, your JVM detects these cores and creates a high amount of threads. {brandname} creates more non-blocking thread pools to manage the threads. These activities consume large amounts of caching resources.

Reply:
Updated to:

> The default setting of `1` thread count indicates that one thread must handle all expired caches by individual order.
Increasing the thread count value means that more threads can handle caches in parallel order.

WDYT?

Reply:
> The default setting of 1 thread count indicates that one thread must handle all expired caches sequentially.

We have no deterministic order.

The other sentence is fine.